DirectProgramming/C++SYCL_FPGA/Tutorials/Features/experimental/annotated_class_clean_coding/src/annotated_class_util.hpp

```diff
@@ -0,0 +1,189 @@
+  static_assert(is_annotated_arg_for_pointer,
+                "'alloc_annotated' cannot be specified with annotated_arg<T> "
+                "as template parameter if T is a non-pointer type");
+};
+
+template <typename T, typename... Props>
+struct get_raw_type<annotated_arg<
+    T *, sycl::ext::oneapi::experimental::detail::properties_t<Props...>>> {
+  using type = T;
+};
+
+// Type traits to get the type of the property list in
+// annotated_arg/annotated_ptr
+template <typename T>
+struct get_property_list {};
+
+template <typename T, typename... Props>
+struct get_property_list<annotated_ptr<
+    T, sycl::ext::oneapi::experimental::detail::properties_t<Props...>>> {
+  using type = sycl::ext::oneapi::experimental::detail::properties_t<Props...>;
+};
+
+template <typename T, typename... Props>
+struct get_property_list<annotated_arg<
+    T, sycl::ext::oneapi::experimental::detail::properties_t<Props...>>> {
+  using type = sycl::ext::oneapi::experimental::detail::properties_t<Props...>;
+};
+
+// Type traits to remove alignment from a property list. This is needed for
+// because the annotated malloc API does not support compile-time alignment
+// property
+template <typename T>
+struct remove_align_from {};
+
+template <>
+struct remove_align_from<
+    sycl::ext::oneapi::experimental::detail::empty_properties_t> {
+  using type = sycl::ext::oneapi::experimental::detail::empty_properties_t;
+};
+
+template <typename Prop, typename... Props>
+struct remove_align_from<
+    sycl::ext::oneapi::experimental::detail::properties_t<Prop, Props...>> {
+  using type = std::conditional_t<
+      sycl::ext::oneapi::experimental::detail::HasAlign<
+          sycl::ext::oneapi::experimental::detail::properties_t<Prop>>::value,
+      sycl::ext::oneapi::experimental::detail::properties_t<Props...>,
+      sycl::ext::oneapi::experimental::detail::merged_properties_t<
+          sycl::ext::oneapi::experimental::detail::properties_t<Prop>,
+          typename remove_align_from<sycl::ext::oneapi::experimental::detail::
+                                         properties_t<Props...>>::type>>;
+};
+
+template <typename T>
+struct split_annotated_type {
+  static constexpr bool is_valid_annotated_type = is_annotated_class<T>::value;
+  static_assert(is_valid_annotated_type,
+                "alloc_annotated function only takes 'annotated_ptr' or "
+                "'annotated_arg' type as a template parameter");
+
+  using raw_type = typename get_raw_type<T>::type;
+  using all_properties = typename get_property_list<T>::type;
+  static constexpr size_t alignment =
+      sycl::ext::oneapi::experimental::detail::GetAlignFromPropList<
+          all_properties>::value;
+  using properties = typename remove_align_from<all_properties>::type;
+};
+
+// Wrapper function that allocates USM host memory with compile-time properties
+// and returns annotated_ptr
+template <typename T>
+T alloc_annotated(size_t count, const sycl::queue &syclQueue,
```

Comment:
The use of camelCase for syclQueue and usmKind seems to violate a consistent naming convention for this file.

Reply:
The variable names have been fixed.